docs/output.md

```diff
@@ -258,6 +258,90 @@ print(result.output)
 
 _(This example is complete, it can be run "as is")_
 
+#### Handling partial output in output functions
+
+!!! warning "Output functions are called multiple times during streaming"
```

Comment:
I'd prefer for this to be top-level text rather than the entire docs section to be an indented warning block